### DISCUSSION TITLE
New extension for generating form elements

### DISCUSSION THREAD

**Discussion by jimustafa** [facelessuser/pymdown-extensions#1554] [Ideas]
When developing a site using markdown, there seems to be a use case for generating form elements for static forms that could use a static form provider. One example would be a Contact Us form, or similar.

Clearly, this can be done with raw HTML, and can even combined with `md_in_html`, but maybe there is still room for a markdown extension. Implementations have been referenced and discussed elsewhere:

- https://talk.commonmark.org/t/introducing-markdown-extensions-for-form-input/432
- https://stackoverflow.com/questions/24051693/input-form-in-markdown

I particularly like the syntax exhibited in https://github.com/brikis98/wmd, which appears to have inspired another implementation https://github.com/slamdata/purescript-markdown.

**Comment by facelessuser** [comment]
Pymdown Extensions came about as a collection of extensions that I actively needed for one reason or another. The extensions didn't exist at the time or I didn't like the existing implementations out there. While I occasionally have and do add new extensions, these situations usually must come from a place where there is the direct motivation for me to implement them, e.g. I will use them.

Unfortunately, Pymdown Extensions cannot be the one-stop place for all extensions. I simply do not have the time to maintain every extension that anyone would ever need.  And since my time is limited, I am very picky about what additional burden I pick up in my projects. Pymdown Extensions is not the only open source project that I maintain, so I try to keep my maintenance burden within reasonable limits.

I can certainly see form elements being very useful to certain people, but as I do not currently have a need for such things, at this time, I am not willing to take on the extra burden of developing such an extension.

I can see how this would be useful to others though, so I would encourage interested parties to look into creating an extension and maintaining it themselves.
